src/near-rpc-provider.ts

```diff
@@ -275,17 +282,22 @@ export class NearRpcProvider extends JsonRpcProvider {
     }
   }
 
-  async getCode(addressOrName: string | Promise<string>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
-    const getCodeParams: GetCodeParams = {
+  private _internalCheckOption(params: any, blockTag: BlockTag | Promise<BlockTag>) {
+    if (blockTag === 'latest') {
+      params.finality = 'final'
+    } else {
+      params.block_id = blockTag
+    }
+    return params
+  }
```

Comment:
I think a better name for this method would be something like `setParamsFinalityOrBlockId`. Also, the `blockTag` type should only be `BlockTag` and not Promise.

Reply:
Okay!